lib/jsonapi/authorization/authorizing_processor.rb

```diff
@@ -85,16 +85,49 @@ def authorize_show_relationship
       end
 
       def authorize_show_related_resource
+        # 1. Load the related resource like jsonapi-resources loads them.
+        #    This code is copy-pasted from jsonapi-resources `Processor#related_resource`
+        #    so that we get to loading the resources the same way as JR does.
+        include_directives = params[:include_directives]
         source_klass = params[:source_klass]
         source_id = params[:source_id]
-        relationship_type = params[:relationship_type].to_sym
+        relationship_type = params[:relationship_type]
+        serializer = params[:serializer]
+        fields = params[:fields]
+
+        find_options = {
+            context: context,
+            fields: fields,
+            filters: {},
+            include_directives: include_directives
+        }
+
+        source_resource = source_klass.find_by_key(source_id, context: context, fields: fields)
+
+        resource_set = find_related_resource_set(source_resource,
+                                                 relationship_type,
+                                                 include_directives,
+                                                 find_options)
+
+        resource_set.populate!(serializer, context, find_options)
 
-        source_resource = source_klass.find_by_key(source_id, context: context)
+        # 2. Authorize the showing of related resources. This code is specific to jsonapi-authorization.
 
-        related_resource = source_resource.public_send(relationship_type)
+        # TODO: Figure out better names for the variables used when looping.
+        #       The reason they have bad names is that we don't know what they are.
+        related_resources = resource_set.resource_klasses.flat_map do |_resource_klass, inner_thing|
+          inner_thing.map { |_resource_id, lower_level_thing| lower_level_thing[:resource] }
+        end
+
+        if related_resources.size > 1
+          raise "Unexpectedly more than one related resource in a show_related_resource operation!"
+        end
+
+        related_resource = related_resources.first
```

Comment:
The amount of new code that we have to write here seems quite scary 😅. As far as I can tell, this is the way things should be done even though it means we have to copy _a lot of code_ from `jsonapi-authorization`: 

https://github.com/cerebris/jsonapi-resources/blob/v0.10.7/lib/jsonapi/processor.rb#L141-L163

Reply:
Seems like this PR had the right code which I can use here:

* https://github.com/venuu/jsonapi-authorization/pull/131